### PR TITLE
DATA-6181: PDP dropin initializer configuration

### PR DESCRIPTION
**Pull request by jeff-matthews**
This pull request (PR) adds a new topic that describes the initializer configuration options for the PDP dropin.

It also provides a detailed use case for setting default options on complex products using `models` and `initialData` as scoped in DATA-6181.

It does not provide detailed use cases for any other configuration options, but if a reviewer wants to suggest them, I'm happy to include 😃. I don't think it should block this PR though.

>[!NOTE]
>This topic deviates from the [dropin_Initialization.mdx](https://github.com/commerce-docs/microsite-commerce-storefront/blob/develop/_dropin-templates/dropin-initialization.mdx) template a little. I didn't understand how to include a "big picture" diagram. I also thought that replacing the "step-by-step" section with an actual use case made more sense. My assumption is that we can add new H2 sections for additional use cases in the future if necessary.

### CODE REVIEW

**Review by keharper — APPROVED** [review]

File: src/content/docs/dropins/product-details/pdp-initialization.mdx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  langDefinitions,
+  models,
+});
+```
+
+## Set default product options
+
+When a user navigates to the product detail page (PDP) on your site, you can set certain options as preselected defaults. This use case allows merchandisers to set default options through the Adobe Commerce Admin, which provides a more customized and streamlined shopping experience.
+
+* **Simple products**—Default option selection is supported out-of-the-box through the Admin (`isDefault`). No additional PDP dropin customization is required.
```
Comment (alandana) [inline src/content/docs/dropins/product-details/pdp-initialization.mdx]: I don't understand what you mean by this. A simple product is a simple product. It does not have any options, so there's no need for setting up any default options.

File: src/content/docs/dropins/product-details/pdp-initialization.mdx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+* **Complex products**—Default option selection is _not_ supported out-of-the-box. Instead, you must use [product attributes](https://experienceleague.adobe.com/en/docs/commerce-admin/catalog/product-attributes/product-attributes) and customize the PDP dropin initializers to define which attribute and value is used by default. If no option is defined as default, the first listed option will be preselected by default.
```
Comment (alandana) [inline src/content/docs/dropins/product-details/pdp-initialization.mdx]: This part is not entirely true - "the first listed option will be preselected by default".
This depends on the type of swatch used. Take this as an example -- https://experience-qa.adobe.net/solutions/magento-datalake-storefront-pdp/assets/qa20241016085457/standalone/product.html?sku=VSW12 -- this product does not have any default values, so no color swatch and no size are automatically selected.

The case that you mention is only happening for the dropdown type swatch. But even then, you can opt out of the preselection in the dropdown by providing a placeholder (eg: "Choose an option").

File: src/content/docs/dropins/product-details/pdp-initialization.mdx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+This approach allows merchants to set different default selections for complex products in the Admin. Developers can then customize the PDP dropin to show the default or preselected options.
+
+<Steps>
+
+1. Log in to the Adobe Commerce Admin.
+
+1. Create a custom attribute (`default_options`) to define default options for each product.
+
+1. Ensure that the `default_options` attribute has been [exported](https://experienceleague.adobe.com/en/docs/commerce-merchant-services/saas-data-export/overview) to the Catalog Service.
```
Comment (alandana) [inline src/content/docs/dropins/product-details/pdp-initialization.mdx]: Ensure that it has been exported _**and**_ has visible in PDP role.

**Review by alandana — COMMENTED** [review]
Thanks @jeff-matthews, I added some initial comments.

File: src/content/docs/dropins/product-details/pdp-initialization.mdx (hunk 1 above)
Comment (jeff-matthews) [reply src/content/docs/dropins/product-details/pdp-initialization.mdx]: My mistake @alandana 😅. I'm not sure what I was thinking there. I'll remove it.

File: src/content/docs/dropins/product-details/pdp-initialization.mdx (hunk 3 above)
Comment (jeff-matthews) [reply src/content/docs/dropins/product-details/pdp-initialization.mdx]: @alandana, will you please elaborate on "has visible in PDP role". I'm not sure what that means. I don't think we have any documentation about that.

Comment (jeff-matthews) [reply src/content/docs/dropins/product-details/pdp-initialization.mdx]: Does it refer to [this](https://experienceleague.adobe.com/en/docs/commerce-admin/catalog/products/types/product-create-configurable#step-4-complete-the-basic-settings:~:text=Accept%20the%20default%20Visibility%20setting%20of%20Catalog%2C%20Search.) or something else?

File: src/content/docs/dropins/product-details/pdp-initialization.mdx (hunk 2 above)
Comment (jeff-matthews) [reply src/content/docs/dropins/product-details/pdp-initialization.mdx]: I'll attempt a rewrite and re-request review.

File: src/content/docs/dropins/product-details/pdp-initialization.mdx (hunk 3 above)
Comment (jeff-matthews) [reply src/content/docs/dropins/product-details/pdp-initialization.mdx]: Ohhh, it must be `roles` in the [ProductViewAttribute](https://developer.adobe.com/commerce/services/graphql/catalog-service/products/#productviewattribute-type) type. I'm going to push a change, but please let me know if it's something else.

File: src/content/docs/dropins/product-details/pdp-initialization.mdx [hunk 4]
```diff
@@ -49,13 +49,11 @@ initializers.register(productApi.initialize, {
...
+Default option selection is _not_ supported out-of-the-box for complex products. Instead, you must use [product attributes](https://experienceleague.adobe.com/en/docs/commerce-admin/catalog/product-attributes/product-attributes) and customize the PDP dropin initializers to define which attribute and value is used by default.
 
-* **Complex products**—Default option selection is _not_ supported out-of-the-box. Instead, you must use [product attributes](https://experienceleague.adobe.com/en/docs/commerce-admin/catalog/product-attributes/product-attributes) and customize the PDP dropin initializers to define which attribute and value is used by default. If no option is defined as default, the first listed option will be preselected by default.
-
-This approach allows merchants to set different default selections for complex products in the Admin. Developers can then customize the PDP dropin to show the default or preselected options.
+If no option is defined as default in the Admin for the dropdown input type, the first listed option will be preselected by default. You can opt out of the preselection in the dropdown by providing a placeholder (for example, "Choose an option").
```
Comment (alandana) [inline src/content/docs/dropins/product-details/pdp-initialization.mdx]: Can we leave out this part?
> You can opt out of the preselection in the dropdown by providing a placeholder (for example, "Choose an option").

This is conditioned by a flag that I've recently introduced, but I'd rather keep this separate for now.

**Review by alandana — COMMENTED** [review]
Left another small comment.

File: src/content/docs/dropins/product-details/pdp-initialization.mdx (hunk 4 above)
Comment (jeff-matthews) [reply src/content/docs/dropins/product-details/pdp-initialization.mdx]: No problem. I removed it.

**Comment by jeff-matthews** [comment]
Thanks @alandana! I removed the statement about no default options being set as requested. Does this look good to publish?

**Comment by jeff-matthews** [comment]
I'm wondering if we should move this to the [PDP functions](https://experienceleague.adobe.com/developer/commerce/storefront/dropins/product-details/pdp-functions/) page instead.

**Review by alandana — APPROVED** [review]
Thanks Jeff, looks good.
And no, it wouldn't fit under the PDP functions. Let's leave it as is.

**Pull request merged**